client/lib/wp/sync-handler/index.js

```diff
@@ -32,7 +72,7 @@ export class SyncHandler {
 	 */
 	constructor( handler ) {
 		// expose `syncHandler` global var (dev mode)
-		if ( 'development' === config( 'env' ) ) {
+		if ( 'development' === config( 'env' ) && typeof window !== 'undefined' ) {
```

Comment:
As is, this cache only makes sense for local use and needs to be cleaned when switching users.